libraries/integration/Microsoft.Bot.Builder.Integration.Runtime/Extensions/ServiceCollectionExtensions.cs

```diff
@@ -116,7 +116,7 @@ internal static void AddBotRuntimeStorage(this IServiceCollection services, ICon
             else
             {
                 // If no storage is configured, default to memory storage
-                services.AddSingleton<IStorage, MemoryStorage>();
+                services.AddSingleton<IStorage>(sp => new MemoryStorage());
```

Comment:
why did you find you had to do this?

Reply:
Me, Carlos, Gabo, and Peter all tried to trace down why this was happening, and only in an Azure Function project.  We could find no definitive reason why DI was selecting the wrong constructor.

Reply:
At this point, this is one of two work arounds I can devise.  This, or change MemoryStorage to default the JsonSerializer if a null is provided via DI.  By all appearances, something in Azure Functions is adding the null JsonSerializer.